app/apphandlers/processorAppHandler.go

```diff
@@ -344,15 +361,15 @@ func (processor *ProcessorApp) LegacyStart(ctx context.Context, options *app.Opt
 				clearDB := false
 				if enableProcessor {
 					g.Go(func() error {
-						StartProcessor(ctx, &clearDB, &gatewayDB, &routerDB, &batchRouterDB, &procErrorDB, reportingI, multitenantStats)
+						StartProcessor(ctx, &clearDB, &gatewayDB, &routerDB, &batchRouterDB, &procErrorDB, reportingI, multitenantStats, transientSources)
```

Comment:
Not really related to this PR, just thinking out loud here so feel free to ignore me. I wonder why we even use an `errgroup` with these routines given that they don't return any error at all.

Reply:
it waits for all go routines to finish in [#417](https://github.com/rudderlabs/rudder-server/pull/1869/files#diff-a56cac86f1d08aa5ed9a9f58c33b5f65aa8b34574a018ed4b12d14cc4b190179R417) so even if this specific task doesn't appear to through an error, we still need to wait for it to complete, upon context cancellation.

Reply:
Yeah, looks like a wait group could do the trick. Might be clearer since `errgroup` clearly has `err` in its name it implies that the error is important.